src/UploadImageBehavior.php

```diff
@@ -89,23 +95,21 @@ public function init()
 
         parent::init();
 
-        if ($this->createThumbsOnSave) {
```

Comment:
createThumbsOnSave - this is optional parameter
This option is necessary to disable the thumbs generation.

Reply:
I undestand. But now, if you specify createThumbsOnSave == false and createThumbsOnRequest this part of the code will not be executed. 

I see three cases: 
1. createThumbsOnSave  == true and createThumbsOnRequest  == false. Thumbs are created on save.
2. createThumbsOnSave   == false and createThumbsOnRequest   == true. Thumbs are created on request.
3. createThumbsOnSave   == false and createThumbsOnRequest   == false. Thumbs are never created.

So, if we have current init() method implementation, in second case checking thumbs parameters will not perform, thumbPath and thumbUrl will not filled if emty.

I think, there is no problem there, if these actions will perform everytime, regardless of whether thumbs are creating or not.

Reply:
@Matvik 

It seems to me so more correctly:

```php
if ($this->createThumbsOnSave || $this->createThumbsOnRequest) {
    // validate thumb params ...
}
```

Reply:
Done